addons/hr_holidays/models/hr_leave.py

```diff
@@ -317,6 +269,17 @@ def _auto_init(self):
                            self._table, ['date_to', 'date_from'])
         return res
 
+    @api.onchange('request_hour_from', 'request_hour_to')
+    def _onchange_hours(self):
+        # avoid negative or after midnight
+        self.request_hour_from = min(self.request_hour_from, 23.99)
```

Comment:
And why not ?
```suggestion
        self.request_hour_from = min(self.request_hour_from, 23.999)
```
or even
```suggestion
        self.request_hour_from = min(self.request_hour_from, 23.9999)
```
This way to check the hours seems wrong. 

+ business logic shouldn't be defined in onchange, as this visual tool is not executed outside of the interface context. Use a compute method instead.

Reply:
Hello @tivisse :wave: 

23.99 will convert into 23:59 and 23.999 will convert into 24:00
I think we don't want to set 24.00 in both the hour_from and hour_to

If I create a compute method and set both request_hour_from and request_hour_to as dependencies, the compute method won't be called until I save the form which won't update the field value until the save

and if we keep each compute method for those fields then we couldn't make it concurrent

Can you please tell me how to use the compute method in this scenario?:innocent: 


There is one bug I found while testing my code
if we use 12.999 something that has 3 or more digits after the decimal point then the float_to_time util method will throw an error because of the precision used in the float_round method
https://github.com/odoo/odoo/blob/17.0/addons/resource/models/utils.py#L45

if you say then we could set precision_digits like 4 or higher maybe

Reply:
I get your point for the 23.999, thanks :)

For the second point, I don't see why a compute method would be less reactive than an onchange, maybe there is a trigger issue in that case ?

Thanks for your answer.

Reply:
Hello @tivisse :wave: 

Yes it's more of the triggering issue I've tried something like this but it won't trigger the compute method until i save the form

    @api.depends('request_hour_to')
    def _compute_request_hour_from(self):
        for record in self:
            record.request_hour_to = min(max(record.request_hour_to, 0.0), 24.0)
            record.request_hour_from = min(max(record.request_hour_from, 0.0), 23.99)
            if record.request_hour_from > record.request_hour_to:
                record.request_hour_from = record.request_hour_to`

```
@api.depends('request_hour_from')
    def _compute_request_hour_to(self):
        for record in self:
            record.request_hour_from = min(max(record.request_hour_from, 0.0), 23.99)
            record.request_hour_to = min(max(record.request_hour_to, 0.0), 24.0)
            if record.request_hour_to < record.request_hour_from:
                record.request_hour_to = record.request_hour_from
```